### ISSUE TITLE
Adding a Jakarta validation annotation. 

### ISSUE THREAD

**Issue by DempseyRoller** [joelittlejohn/jsonschema2pojo#1524]
I'd like to use @NotBlank annotation with groups i.e. @NotBlank(groups = Someclass.class) is there a way to implement this into jsonschema2pojo? The customannotator perhaps?

**Comment by joelittlejohn** [comment]
Could you achieve the same with `pattern`? You could use a regex like `.+`.

**Comment by DempseyRoller** [comment]
But can you use group validation then? https://www.baeldung.com/java-validation-groups this is what I'm getting at. The project I'm inheriting needs to use one base schema for about six different events some of which don't use the same fields, but if it has a field I should validate that it's populated and is valid. It means that I can't just put every thing to required as then most of the requests wouldn't be valid.

**Comment by DempseyRoller** [comment]
I managed to do it with NotNull by using a custom rulefactory and borrowing a bit of your RequiredRule code. The instructions how to use a rulefactory or an annotator are quite sparse, so it might not be optimal. But hey it works and it's not bad if it's a bit home made when it only runs for the generation.

**Comment by unkish** [comment]
> But hey it works and it's not bad if it's a bit home made when it only runs for the generation.

Glad to hear that you have managed to sort it out.

> The instructions how to use a rulefactory or an annotator are quite sparse, so it might not be optimal

Anything particular that would be worth mentioning ?


I suppose that given issue may be closed then ?

**Comment by DempseyRoller** [comment]
I think there could be a small basic example of using a customannotator and a custom rulefactory in the read me or on the site. Now I found the possibility of even having them after many Google searchs and the maven searching the maven plugin docs. It's not too easy to find how the generation works through the sources code, although I think I got it in the end.

Otherwise the library is very nice.
